src/proxy/inbound.rs

```diff
@@ -259,138 +291,170 @@ impl Inbound {
         metrics: Arc<Metrics>,
         socket_factory: Arc<dyn SocketFactory + Send + Sync>,
     ) -> Result<Response<Empty<Bytes>>, hyper::Error> {
-        match req.method() {
-            &Method::CONNECT => {
-                let uri = req.uri();
-                info!("got {} request to {}", req.method(), uri);
-                let addr: Result<SocketAddr, _> = uri.to_string().as_str().parse();
-                if addr.is_err() {
-                    info!("Sending 400, {:?}", addr.err());
-                    return Ok(Response::builder()
-                        .status(StatusCode::BAD_REQUEST)
-                        .body(Empty::new())
-                        .unwrap());
-                }
+        let method = req.method();
+        if req.method() != Method::CONNECT {
+            info!("Sending 404, got {method}");
+            return Ok(Response::builder()
+                .status(StatusCode::NOT_FOUND)
+                .body(Empty::new())
+                .unwrap());
+        }
+        let uri = req.uri();
+        info!("got {} request to {}", req.method(), uri);
+        let addr: Result<SocketAddr, _> = uri.to_string().as_str().parse();
+        if addr.is_err() {
+            info!("Sending 400, {:?}", addr.err());
+            return Ok(Response::builder()
+                .status(StatusCode::BAD_REQUEST)
+                .body(Empty::new())
+                .unwrap());
+        }
 
-                let addr: SocketAddr = addr.unwrap();
-                if addr.ip() != conn.dst.ip() {
-                    info!("Sending 400, ip mismatch {addr} != {}", conn.dst);
-                    return Ok(Response::builder()
-                        .status(StatusCode::BAD_REQUEST)
-                        .body(Empty::new())
-                        .unwrap());
-                }
-                // Orig has 15008, swap with the real port
-                let conn = Connection { dst: addr, ..conn };
-                let dst_network_addr = NetworkAddress {
-                    network: conn.dst_network.to_string(), // dst must be on our network
-                    address: addr.ip(),
-                };
-                let Some((upstream, upstream_service)) =
-                    state.fetch_workload_services(&dst_network_addr).await
-                else {
-                    info!(%conn, "unknown destination");
-                    return Ok(Response::builder()
-                        .status(StatusCode::NOT_FOUND)
-                        .body(Empty::new())
-                        .unwrap());
-                };
-                let has_waypoint = upstream.waypoint.is_some();
-                let from_waypoint = Self::check_waypoint(state.clone(), &upstream, &conn).await;
-                let from_gateway = Self::check_gateway(state.clone(), &upstream, &conn).await;
+        let addr: SocketAddr = addr.unwrap();
```

Comment:
**nitty**: this looks a bit odd. I think you can `let addr = match addr{...}` and it works Ok so long as arms that won't result in the desired type return.